snowpack/src/sources/local.ts

```diff
@@ -202,125 +345,111 @@ export default {
     installOptions.polyfillNode = config.packageOptions.polyfillNode;
     installOptions.packageLookupFields = config.packageOptions.packageLookupFields;
     installOptions.packageExportLookupFields = config.packageOptions.packageExportLookupFields;
+    if (config.packageOptions.source !== 'local') {
+      installOptions.cwd = this.packageRootDirectory;
+      this.lockfile = (await readLockfile(this.config.root)) || this.lockfile;
+      await setupPackageRootDirectory(this.packageRootDirectory, this.lockfile);
+    const buildArb = new Arborist({
+      ...(typeof config.packageOptions.source === 'string' ? {} : config.packageOptions.source),
+      path: PKG_SOURCE_DIR,
+    });
+      await buildArb.buildIdealTree();
+      await buildArb.reify();
+    }
     return installOptions;
-  },
-
-  // TODO: in build+watch, run prepare()
-  //  then, no import map
-  //
+  }
 
-  async prepare(commandOptions: CommandOptions) {
-    config = commandOptions.config;
-    const DEV_DEPENDENCIES_DIR = path.join(
-      PROJECT_CACHE_DIR,
-      process.env.NODE_ENV || 'development',
-    );
-    const installDirectoryHashLoc = path.join(DEV_DEPENDENCIES_DIR, '.meta');
-    const installDirectoryHash = await fs
-      .readFile(installDirectoryHashLoc, 'utf-8')
-      .catch(() => null);
-    if (installDirectoryHash === 'v1') {
-      logger.debug(`Install directory ".meta" tag is up-to-date. Welcome back!`);
-      return;
-    } else if (installDirectoryHash) {
-      logger.info(
-        'Snowpack updated! Rebuilding your dependencies for the latest version of Snowpack...',
-      );
-    } else {
-      logger.info(
-        `${colors.bold('Welcome to Snowpack!')} Because this is your first time running\n` +
-          `this project${
-            process.env.NODE_ENV === 'test' ? ` (mode: test)` : ``
-          }, Snowpack needs to prepare your dependencies. This is a one-time step\n` +
-          `and the results will be cached for the lifetime of your project. Please wait...`,
-      );
+  private async installPackage(packageName: string, source: string): Promise<boolean> {
+    const {config, arb} = this;
+    if (config.packageOptions.source === 'local') {
+      return false;
     }
-    const installTargets = await getInstallTargets(config, config.packageOptions.knownEntrypoints);
-    if (installTargets.length === 0) {
-      logger.info('No dependencies detected. Ready!');
-      return;
+    const lookupStr = path.relative(this.packageRootDirectory, source);
+    const lookupParts = lookupStr.split(path.sep);
+    let lookupNode = arb.actualTree || arb.virtualTree;
+    let exactLookupNode = lookupNode;
+    while (lookupNode && lookupNode.children && lookupParts.length > 0) {
+      const part = lookupParts.shift();
+      if (part !== 'node_modules') {
+        continue;
+      }
+      let lookupPackageName = lookupParts.shift()!;
+      if (lookupPackageName.startsWith('@')) {
```

Comment:
This might need to be a utility. Even if only within this file (I see it used in at least one other place here).

Reply:
Oh nice, good catch. Meant to pull that into a util

Reply:
done